srcbpatch/coloredconsole.cpp

```diff
@@ -0,0 +1,96 @@
+#include "stdafx.h"
+#include <algorithm>
+#include "coloredconsole.h"
+
+#if !defined(__linux__) && !((defined(__APPLE__) && defined(__MACH__)))
+#include <windows.h>
+#endif
+
+
+namespace coloredconsole
+{
+#if !defined(__linux__) && !((defined(__APPLE__) && defined(__MACH__)))
+    /// <summary>
+    ///   RAII for set console output color in windows
+    /// </summary>
+    class WindowsConsoleColor final
+    {
+        WindowsConsoleColor(const WindowsConsoleColor&) = delete;
+        WindowsConsoleColor(WindowsConsoleColor&&) = delete;
+        WindowsConsoleColor& operator =(const WindowsConsoleColor&) = delete;
+        WindowsConsoleColor& operator =(WindowsConsoleColor&&) = delete;
```

Comment:
I counted 4 matches for this. May be create some class with deleted methods and use private inheritance?